### DISCUSSION TITLE
Setting up Bitbucket Cloud - "Error committing files"

### DISCUSSION THREAD

**Discussion by mikemfleming** [renovatebot/renovate#7948] [Request Help] [answered]
### Which Renovate are you using?
Renovate Open Source CLI

### Which platform are you using?
Bitbucket Cloud

### Have you checked the logs? Don't forget to include them if relevant
This is the command I run
```
renovate --log-level=debug \
    --password=$PASSWORD \
    --username=$USERNAME \
    --platform=bitbucket \
    some-org/some-repo
```
And this is the error I get
```
 INFO: Renovate is exiting with a non-zero code due to the following logged errors
       "loggerErrors": [
         {
           "name": "renovate",
           "level": 50,
           "logContext": "tRNuW66eR",
           "repository": "$NPM_REGISTRY_USER/$NPM_REGISTRY_USER-front-end",
           "err": {
             "task": {
               "commands": [
                 "push",
                 "origin",
                 "renovate/configure:renovate/configure",
                 "--force",
                 "-u",
                 "--no-verify",
                 "--verbose",
                 "--porcelain"
               ],
               "format": "utf-8"
             },
             "message": "Pushing to https://**redacted**@bitbucket.org/$NPM_REGISTRY_USER/$NPM_REGISTRY_USER-front-end.git\nPOST git-receive-pack (566 bytes)\nremote: error: cannot lock ref 'refs/heads/renovate/configure': 'refs/heads/renovate' exists; cannot create 'refs/heads/renovate/configure'        \nerror: failed to push some refs to 'https://**redacted**@bitbucket.org/$NPM_REGISTRY_USER/$NPM_REGISTRY_USER-front-end.git'\n",
             "stack": "Error: Pushing to https://**redacted**@bitbucket.org/$NPM_REGISTRY_USER/$NPM_REGISTRY_USER-front-end.git\nPOST git-receive-pack (566 bytes)\nremote: error: cannot lock ref 'refs/heads/renovate/configure': 'refs/heads/renovate' exists; cannot create 'refs/heads/renovate/configure'        \nerror: failed to push some refs to 'https://**redacted**@bitbucket.org/$NPM_REGISTRY_USER/$NPM_REGISTRY_USER-front-end.git'\n\n    at GitExecutorChain.onFatalException (/usr/local/share/.config/yarn/global/node_modules/simple-git/src/lib/runners/git-executor-chain.js:60:87)\n    at GitExecutorChain.<anonymous> (/usr/local/share/.config/yarn/global/node_modules/simple-git/src/lib/runners/git-executor-chain.js:51:28)\n    at Generator.throw (<anonymous>)\n    at rejected (/usr/local/share/.config/yarn/global/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)"
           },
           "msg": "Error committing files."
         },
         {
           "name": "renovate",
           "level": 50,
           "logContext": "tRNuW66eR",
           "repository": "$NPM_REGISTRY_USER/$NPM_REGISTRY_USER-front-end",
           "err": {
             "task": {
               "concatStdErr": false,
               "format": "utf-8",
               "commands": ["rev-parse", "origin/renovate/configure"]
             },
             "message": "fatal: ambiguous argument 'origin/renovate/configure': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'\n",
             "stack": "Error: fatal: ambiguous argument 'origin/renovate/configure': unknown revision or path not in the working tree.\nUse '--' to separate paths from revisions, like this:\n'git <command> [<revision>...] -- [<file>...]'\n\n    at GitExecutorChain.onFatalException (/usr/local/share/.config/yarn/global/node_modules/simple-git/src/lib/runners/git-executor-chain.js:60:87)\n    at GitExecutorChain.<anonymous> (/usr/local/share/.config/yarn/global/node_modules/simple-git/src/lib/runners/git-executor-chain.js:51:28)\n    at Generator.throw (<anonymous>)\n    at rejected (/usr/local/share/.config/yarn/global/node_modules/simple-git/src/lib/runners/git-executor-chain.js:6:65)\n    at runMicrotasks (<anonymous>)\n    at processTicksAndRejections (internal/process/task_queues.js:97:5)"
           },
           "msg": "Repository has unknown error"
         }
       ]
```

### What would you like to do?

I'm trying to run set up renovate to check our dependencies and run in our Pipelines. I initially followed up [here](https://github.com/renovatebot/config-help/issues/355).

Is it possible that I've set up my Bitbucket user with the wrong permissions?

**Comment by viceice** [comment]
1. Did you added /replaced this  `$NPM_REGISTRY_USER/$NPM_REGISTRY_USER` in / to the logs? Looks like you've made a config issue?

1. Did you checked out our brand new [config samples](https://github.com/renovatebot/docker-renovate/blob/master/docs/bitbucket.md) for bitbucket?

1. Please shares as much as possible configuration as possible, so we can check it for mistakes.

**Comment by label-actions** [comment]
Please add your Renovate config to this issue.

**Comment by mikemfleming** [comment]
Thanks for linking to the config samples. Also, apologies for replying to that old issue that I linked to! Didn't see your message there.

> Did you added /replaced this $NPM_REGISTRY_USER/$NPM_REGISTRY_USER in / to the logs?
That is the raw output in the console, I thought it may have been redacted by renovate but I guess not. We have some private npm packages that we set up with those variables. Do you think if I ignored those dependencies in `renovate.json` this command would succeed?

I updated how I'm running renovate based on the config samples and it results in the original error. Here is the config info you requested:

`pipelines`
```
    - step:
        name: renovate
        image: renovate/renovate:23.96.2-slim
        services:
          - docker
        caches:
          - docker
        script:
          - export LOG_LEVEL=debug RENOVATE_CONFIG_FILE="$BITBUCKET_CLONE_DIR/renovate.config.js"
          - renovate
```
`renovate.json`
```
{
  "extends": [
    "config:base"
  ],
  "prConcurrentLimit": 1
}
```
`renovate.config.js`
```
module.exports = {
    platform: 'bitbucket',
    username: process.env.RENOVATE_USER,
    password: process.env.RENOVATE_SECRET,
    baseDir: `${process.env.BITBUCKET_CLONE_DIR}/renovate`,
    repositories: [ "some-org/some-repo"],
  }
```

**Comment by viceice** [comment]
Looks like bitbucket.org has a broken repo. can you try to manually push a `renovate/configure` branch?

**Comment by viceice** [comment]
I've seen `remote: error: cannot lock ref 'refs/heads/renovate/configure'` on my self-hosted gitlab some time ago, and only could fix the repo on gitlab server. Was a diffrerent ref, but same error.

**Comment by mikemfleming** [comment, marked as answer]
I figured it out! Your configuration is right on but it may be worth mentioning somewhere to _not_ name the branch your working on `renovate` as that is what renovate is trying to work off of. 

I fixed the issue by renaming the branch to something else and deleting it on the origin. 

Thank you so much for your help! 

🙌

**Comment by rarkins** [comment]
@viceice maybe we can add an init check in the repo to make sure that there is not an existing branch that blocks the branchPrefix? i.e. `renovate` in the case of `branchPrefix=renovate/`. We could raise it as a config error.

**Reply by viceice** [reply]
@rarkins sure, good idea. can you open an issue to track this?

**Reply by HonkingGoose** [reply]
> 
> 
> @viceice maybe we can add an init check in the repo to make sure that there is not an existing branch that blocks the branchPrefix? i.e. `renovate` in the case of `branchPrefix=renovate/`. We could raise it as a config error.

The issue tracking this work can be found at: #7968.
